lib/std/time.zig

```diff
@@ -344,6 +344,221 @@ test Timer {
+                    .minute = std.math.comptimeMod(m, 60),
+                    .hour = std.math.comptimeMod(h, 24),
+                },
+                .day_overflow = overflow,
+            };
+        }
+
+        /// Does not handle leap seconds.
+        pub fn add(self: Self, hour: i64, minute: i64, second: i64, subsecond: i64) Self {
+            return self.addWithOverflow(hour, minute, second, subsecond).time;
+        }
+
+        pub fn fromRfc3339(str: []const u8) !Self {
+            if (str.len < "hh:mm:ssZ".len) return error.Parsing;
+
+            if (str[2] != ':' or str[5] != ':') return error.Parsing;
+
+            const hour = try std.fmt.parseInt(Hour, str[0..2], 10);
+            const minute = try std.fmt.parseInt(Minute, str[3..5], 10);
+            const second = try std.fmt.parseInt(Second, str[6..8], 10);
+
+            var i: usize = 8;
+            const subsecond: Subsecond = if (str[i] == '.') brk: {
+                i += 1;
+                while (i < str.len and std.ascii.isDigit(str[i])) : (i += 1) {}
+                if (Subsecond == u0) break :brk 0;
+                const subsecond_str = str[9..i];
+                // Choose largest performant type.
+                // Ideally, this would allow infinite precision.
+                const T = f64;
+                var subsecond = try std.fmt.parseFloat(T, subsecond_str);
+                const actual_precision: T = @floatFromInt(subsecond_str.len);
+                subsecond *= std.math.pow(T, 10, precision - actual_precision);
+
+                break :brk @intFromFloat(subsecond);
+            } else 0;
+
+            // timezone required
+            if (str.len <= i) return error.Parsing;
+
+            const offset = if (std.ascii.toUpper(str[i]) == 'Z') 0 else brk: {
+                var sign: IMinutes = 1;
+                if (str[i] == '-') {
+                    sign = -1;
+                    i += 1;
+                } else if (str[i] == '+') {
+                    i += 1;
+                }
+
+                const offset_hour = try std.fmt.parseInt(IMinutes, str[i..][0..2], 10);
+                if (str[i + 2] != ':') return error.Parsing;
+                const offset_minute = try std.fmt.parseInt(IMinutes, str[i + 3 ..][0..2], 10);
+
+                break :brk sign * (offset_hour * 60 + offset_minute);
+            };
+
+            return .{ .hour = hour, .minute = minute, .second = second, .subsecond = subsecond, .offset = offset };
+        }
+
+        pub fn toRfc3339(self: Self, writer: anytype) !void {
+            if (self.hour > 24 or self.minute > 59 or self.second > 60) return error.Range;
+            try writer.print("{d:0>2}:{d:0>2}:{d:0>2}", .{ self.hour, self.minute, self.second });
+            if (self.subsecond != 0) {
+                // We could trim trailing zeros here to save space.
+                try writer.print(".{d}", .{self.subsecond});
+            }
+            if (self.offset == 0) {
```

Comment:
an offset of zero could also be from a timezone (e.g. GMT), not necessarily UTC.

Reply:
RFC 3339 says `Z` and `[+-]00:00` are equivalent:

```
      Z           A suffix which, when applied to a time, denotes a UTC
                  offset of 00:00; often spoken "Zulu" from the ICAO
                  phonetic alphabet representation of the letter "Z".
```

Reply:
ok, thanks for clarifying.